### PR TITLE
Gang PodGroup Doc

### PR DESCRIPTION
**Pull request by umialpha**
New feature Implementation, Gang PodGroup.

### CODE REVIEW

File: docs/design/gang-podgroup.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# Gang PodGroups
+
+[@umiaplha](http://github.com/umialpha); Feb 20, 2020
+
+## Motivation
+Pods are grouped by PodGroup, but what if we need to group `PodGroups`?
+For example, Job1{taskNum: 3, MinMember: 2}, Job2{taskNum:3, MinMember:2}, Job1 must co-run with Job2.
```
Comment (k82cn) [inline docs/design/gang-podgroup.md]: I'm thinking we only add more info to each taskSpec in Job.

File: docs/design/gang-podgroup.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+Currently, we cannot handle such case.
+Thus, a new feature named `Gang PodGroups` is introduced.
+
+## Function Detail
+A type of string "SubGroup" is introduced into the PodGroupSpec:
+```go
+type PodGroupSpec struct {
+	// SubGroups defines the groups of PodGroup.
+	// When PodGroups grouped together, they are scheduled as a whole.
+	SubGroup string
```
Comment (k82cn) [inline docs/design/gang-podgroup.md]: Prefer to add `SubGroupSpec` for its attributes, e.g. `minMember`. It's complex to reconstruct PodGroup and its SubGroup in scheduler.

File: docs/design/gang-podgroup.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+```
+
+### Allocate Action
+Currenly the allocation flow is as follows,
+
+``` go
+// pseudocode
+namespace := Namepaces.Pop()
```
Comment (k82cn) [inline docs/design/gang-podgroup.md]: It's better to describe the steps without code.

File: docs/design/gang-podgroup.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
+# Gang PodGroups
```
Comment (k82cn) [inline docs/design/gang-podgroup.md]: Prefer to `SubGroups of PodGroup`, or other similar title; as this feature not only used by gang in future.

File: docs/design/gang-podgroup.md (hunk 2 above)
Comment (umialpha) [reply docs/design/gang-podgroup.md]: > Prefer to add `SubGroupSpec` for its attributes, e.g. `minMember`. It's complex to reconstruct PodGroup and its SubGroup in scheduler.

 `SubGroup` is  just like `Namespace` or `Queue`, specifying the SubGroup Name it belongs.

I have completed the code, shall I send a new PR for the detail?
Or we can talk more about your idea, maybe I misunderstand your idea.

Comment (k82cn) [reply docs/design/gang-podgroup.md]: let's add an item to volcano weekly meeting or book another meeting for that :)

Comment (umialpha) [reply docs/design/gang-podgroup.md]: > let's add an item to volcano weekly meeting or book another meeting for that :)

Great, what is the time and link about the meeting this week? I would like to audit :)

Comment (k82cn) [reply docs/design/gang-podgroup.md]: There's a weekly meeting at 14:30, every Thursday (Beijing Time), and here's the agenda of Volcano weekly meeting: https://docs.google.com/document/d/1YLbF8zjZBiR9PbXQPB22iuc_L0Oui5A1lddVfRnZrqs/edit?usp=sharing

**Comment by hzxuzhonghu** [comment]
ping @umialpha

**Comment by umialpha** [comment]
Maybe we could discuss this request again and clarify the details?

**Comment by k82cn** [comment]
> Maybe we could discuss this request again and clarify the details?

That's great ! We can discuss this at Volcano weekly meeting :)

**Pull request closed without merging**